contracts/src/Gateway.sol

```diff
@@ -76,19 +77,27 @@ contract Gateway is IGateway, IInitializable {
         address agentExecutor,
         uint256 dispatchGas,
         ParaID bridgeHubParaID,
-        bytes32 bridgeHubHubAgentID,
+        bytes32 bridgeHubAgentID,
         ParaID assetHubParaID,
-        bytes32 assetHubHubAgentID,
+        bytes32 assetHubAgentID,
         bytes2 createTokenCallID
     ) {
+        if (
+            dispatchGas == 0 || bridgeHubParaID == ParaID.wrap(0) || bridgeHubAgentID == 0
+                || assetHubParaID == ParaID.wrap(0) || assetHubAgentID == 0 || bridgeHubParaID == assetHubParaID
+                || bridgeHubAgentID == assetHubAgentID
+        ) {
```

Comment:
Unfortunately this is the formatting from `forge fmt`.

Reply:
Maybe this check can also be done to DeployScript.sol to catch any issues early in deployment.